src/Microsoft.OData.Core/JsonLight/ODataJsonLightReader.cs

```diff
@@ -2571,8 +2581,8 @@ private sealed class JsonLightResourceSetScope : Scope
             ///   it's the expected base type of the entries in the resource set.
             ///   note that it might be a more derived type than the base type of the entity set for the resource set.
             /// In all cases the specified type must be an entity type.</remarks>
-            internal JsonLightResourceSetScope(ODataResourceSetBase resourceSet, IEdmNavigationSource navigationSource, IEdmStructuredType expectedResourceType, SelectedPropertiesNode selectedProperties, ODataUri odataUri, bool isDelta)
-                : base(isDelta ? ODataReaderState.DeltaResourceSetStart : ODataReaderState.ResourceSetStart, resourceSet, navigationSource, expectedResourceType, odataUri)
+            internal JsonLightResourceSetScope(ODataResourceSetBase resourceSet, IEdmNavigationSource navigationSource, IEdmStructuredType expectedResourceType, SelectedPropertiesNode selectedProperties, ODataUri odataUri, bool isDelta, bool resourceNullable = false)

```

Comment:
tag for resourceNullable needs to be added.
Also, looks like this cstor is used only twice in this file, can we make it not optional by updating the two references?

Reply:
I believe making it optional conveys that by default, the resourcetype is nullable if not specified.